std-bits/base/src/main/java/org/enso/base/enso_cloud/telemetry/TelemetryLogMessage.java

```diff
@@ -0,0 +1,44 @@
+package org.enso.base.enso_cloud.telemetry;
+
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.fasterxml.jackson.databind.node.TextNode;
+import java.util.Objects;
+import org.enso.base.enso_cloud.logging.LogMessage;
+
+public final class TelemetryLogMessage extends LogMessage {
+
+  private final String loggerName;
+  private final ObjectNode extraMeta;
+
+  private TelemetryLogMessage(String message, String loggerName, ObjectNode extraMeta) {
+    super(message);
+    this.loggerName = loggerName;
+    this.extraMeta = extraMeta;
+  }
+
+  public static TelemetryLogMessage create(String loggerName, String message, ObjectNode metadata) {
+    Objects.requireNonNull(loggerName);
+    Objects.requireNonNull(message);
+    Objects.requireNonNull(metadata);
+    return new TelemetryLogMessage(message, loggerName, metadata);
+  }
+
+  @Override
+  protected String kind() {
+    return "Telemetry";
+  }
+
+  @Override
+  protected ObjectNode extraPayload() {
```

Comment:
If this was a public API, I'd prefer `Map<String,Object>` to re-exporting Jackson API. If this is hidden/used only internally it doesn't matter.

Reply:
It is internal only, visible to classes in package `org.enso.base.enso_cloud.logging`.

Reply:
Btw. if it is used by classes in package `org.enso.base.enso_cloud.logging` only, then it doesn't have to be `protected` - being _package private is enough_.